src/nublado2/imageinfo.py

```diff
@@ -0,0 +1,76 @@
+from dataclasses import dataclass, field
+from typing import Dict
+
+FIELD_DELIMITER = "|"
+
+# This type is the dict that comes from cachemachine:
+#  https: //github.com/lsst-sqre/cachemachine/blob/4802ab7d78aec27b400f66b9af3324180910476e/src/cachemachine/types.py#L50-L69  # noqa: E501
+CachemachineEntry = Dict[str, str]
+
+
+@dataclass(frozen=True)
+class ImageInfo:
+    """reference is a docker image reference.
+    See:
+     https://github.com/distribution/distribution/blob/main/reference/reference.go  # noqa: E501
+
+    Example: "registry.hub.docker.com/lsstsqre/sciplat-lab:w_2021_13"
+    """
+
+    reference: str
+
+    """display_name is a human-readable description of the image.
+
+    Example: "Weekly 13"
+    """
+    display_name: str
+
+    """digest is the hash of the last layer of the docker container.  If
+    unknown, it will be the empty string.
+
+    Example: "sha256:419c4b7e14603711b25fa9e0569460a753c4b2449fe275bb5f89743b01794a30"  # noqa: E501
+    """
+    digest: str
+
+    """packed_string is the form in which the image info is packed into the
+    JupyterHub options form and in which is is returned as the form
+    selection.  It is specification, display_name, and digest
+    concatenated with the pipe character.
+
+    Example: "registry.hub.docker.com/lsstsqre/sciplat-lab:w_2021_13|Weekly 13|sha256:419c4b7e14603711b25fa9e0569460a753c4b2449fe275bb5f89743b01794a30"  # noqa: E501
+    """
+    packed_string: str = field(init=False, default="")
+
+    @classmethod
+    def from_cachemachine_entry(cls, entry: CachemachineEntry) -> "ImageInfo":
+        """Take an entry from a cachemachine response, and set our fields
+        from it.
+        """
+        # entry uses None, we use empty string for missing digest
+        return ImageInfo(
```

Comment:
```suggestion
        return cls(
```

Same this, but it helps in case there's some subclassing going on.

Reply:
OK